machine.cpp

```diff
@@ -1,449 +0,0 @@
-/*
- * machine.cpp
- *
- * Baker, Ballard, Jager-Kujawa
- * CSC 341
- * Spring 2016
- */
-
-#include "machine.hpp"
-
-// Initialize machine registers
-registers machine = {0, 0, 0, 0, 0, 0, 0};
-
-unsigned short int main_memory[256];
```

Comment:
Any references to `main_memory` (specifically, the index passed to `main_memory`) need to be modified so the address is translated from logical to physical.

The question is: should the MMU be implemented in hardware, with a reference to a page table? Or should the MMU be in software, making `machine` no longer self-enclosed?